[O11y][Cassandra] Add documentation for Jolokia setup (#9817)
elastic/integrations: packages/cassandra/_dev/build/docs/README.md

This integration has been tested against `Cassandra version 3.11.11`.

## Steps to Setup Jolokia

### Prerequisites

- Java Development Kit (JDK) 1.8 or later
- Apache Cassandra 3.x/4.x (Adjust according to version)
- Jolokia agent

### Jolokia Setup

> To monitor Cassandra with Jolokia, need to set up the Jolokia JVM agent.

Reviewer: ```suggestion
Follow these steps to set up Jolokia for monitoring Apache Cassandra:
```

1. Download the Jolokia JVM Agent:

   Obtain the latest Jolokia JVM agent `.jar` file from the [Jolokia official download page](https://repo1.maven.org/maven2/org/jolokia/jolokia-jvm/).

2. Place the Jolokia Agent:

   Copy the downloaded `.jar` file to a suitable location on the server where Cassandra is installed.

   For example:

   ```bash
   sudo cp jolokia-jvm-<version>-agent.jar /path/to/cassandra/lib/
   ```

> 

Reviewer: ```suggestion
Replace `/path/to/cassandra/lib/` with the actual path to Cassandra's library directory.

```

Author: Updated, thanks!

3. Configure Cassandra to use the Jolokia Agent:

>    To enable the Jolokia agent, modify the `cassandra-env.sh` by adding the following line at the bottom of file:

Reviewer: ```suggestion
   Open the `cassandra-env.sh` file, located in the Cassandra configuration directory, using a text editor, and add the following line at the bottom of the file:
```

   ```
   JVM_OPTS="$JVM_OPTS -javaagent:/path/to/jolokia-jvm-<version>-agent.jar=port=7777,host=0.0.0.0"
   ```

> 

Reviewer: ```suggestion
Replace `/path/to/jolokia-jvm-<version>-agent.jar` with the actual path to the Jolokia agent JAR file copied in Step 2. Save the changes and close the `cassandra-env.sh` file.

```

Author: Updated, thanks!

> 4. Restart Cassandra:
> 
>    After configuring Cassandra with the Jolokia agent, restart the Cassandra service to apply the changes.

Reviewer: ```suggestion
4. Restart Cassandra:

   Restart the Apache Cassandra service to apply the changes made to the configuration.
```

>    Note:
>    - Restarting the Cassandra service will temporarily disrupt database connectivity. Dependent services may experience failures and should be designed to handle such interruptions gracefully.
>    - Immediately after a restart, Cassandra's performance may be impacted due to cold caches and commit log replay.
>    - Ensure no cluster maintenance tasks are in progress during the restart to prevent any unintended consequences.

Reviewer: ```suggestion
  > Note:
 - Restarting the Apache Cassandra service will temporarily disrupt database connectivity. Ensure that dependent services are designed to handle such interruptions gracefully.
 - Immediately after a restart, Cassandra's performance may be impacted due to cold caches and commit log replay. Allow some time for the system to stabilize.
 - Before restarting Cassandra, ensure that no cluster maintenance tasks are in progress to prevent any unintended consequences.
```

Reviewer 2: Please note the exact steps will vary based on the installation type. For example: if cassandra is installed via kkubernetes, the steps above could be different,. 

We could mention, this is representative example only and can provide minimal guideline on restart scenario.

> 

Reviewer: Do you think we add something like:

**Verifying the setup**

After restarting Cassandra, you can verify that Jolokia is properly set up by accessing the Jolokia endpoint:

```
http://<cassandra-host>:7777/jolokia/
```

Replace <cassandra-host> with the hostname or IP address of your Cassandra server.

If the setup is successful, you should see a JSON response containing information about the available Jolokia operations and the Cassandra instance.

You can now use Jolokia to monitor and manage your Apache Cassandra cluster.

Reviewer 2: Agree, this step will be useful.

Author: Updated, thanks!
